File: backend/src/main/java/io/github/bernardusz/cms/comment/Comment.java
package io.github.bernardusz.cms.comment;

import java.time.LocalDateTime;

public record Comment(
    Long id,
    String title,
    String content,
    LocalDateTime createdAt,
    LocalDateTime edited,
    Long likesCount,
    Long dislikesCount,
    Long userId,
    Long contentId
) { }

File: frontend/src/app/types/comment.ts
export interface CommentUpdate {
    title: string;
    content: string;
}

export interface CommentCreation {
    title: string;
    content: string;
    userId: number;
    contentId: number;
}

export interface CommentDetail {
    id: number;
    title: string;
    content: string;
    createdAt: Date;
    edited: Date;
    likesCount: number;
    dislikesCount: number;
    userId: number;
    contentId: number;
    alreadyLiked: boolean;
    alreadyDisliked: boolean;
}

export interface Comment {
    id: number;
    title: string;
    content: string;
    createdAt: Date;
    edited: Date;
    likesCount: number;
    dislikesCount: number;
    userId: number;
    contentId: number;
}
